src/app/components/elements/panels/EventGenderDetails.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { EventBookingDTO, UserRole } from "../../../../IsaacApiTypes";
+import { Col, Table, UncontrolledTooltip } from "reactstrap";
+import { asPercentage } from "../../../services";
+
+export const countEventDetailsByRole = (role: UserRole | undefined, eventBookings: EventBookingDTO[]) => {
+  const countByRole = {
+    genders: {
+      male: 0,
+      female: 0,
+      other: 0,
+      preferNotToSay: 0,
+      unknown: 0,
+    },
+    numberOfConfirmedOrAttendedBookings: 0,
+  };
+
+  eventBookings.forEach((booking) => {
+    const gender = booking.userBooked?.gender;
+    const bookingStatus = booking.bookingStatus;
+    const userRole = booking.userBooked?.role;
+
+    const roleFilter = role === undefined || userRole === role;
```

Comment:
It looks like this constant could be renamed to something a bit more meaningful.

Reply:
renamed to `meetsRoleRequirements`